### PR TITLE
UCT/DM: enabled DM by default

### PR DESCRIPTION
**Pull request by hoopoepg**
- allocated 1 buffer for 2k length

### CODE REVIEW

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 1]
```diff
@@ -606,7 +606,6 @@ uct_rc_mlx5_txqp_dptr_post(uct_rc_iface_t *iface, enum ibv_qp_type qp_type,
     case MLX5_OPCODE_SEND:
         /* Data segment only */
         ucs_assert(length < (2ul << 30));
-        ucs_assert(length <= iface->super.config.seg_size);
```
Comment (hoopoepg) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: this assert is no more relevant because DM length may be over seg_size

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6562/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
doxygen issue

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6563/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6568/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6570/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6573/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
failed due to https://github.com/openucx/ucx/issues/2417

**Comment by hoopoepg** [comment]
bot:mlx:retest

File: src/uct/ib/rc/accel/rc_mlx5_common.h (hunk 1 above)
Comment (brminich) [reply src/uct/ib/rc/accel/rc_mlx5_common.h]: need to revert it back. We can't send more than seg size

Comment (hoopoepg) [reply src/uct/ib/rc/accel/rc_mlx5_common.h]: right, reverted

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6575/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6579/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg how did the tests pass without a logger fix?

**Comment by hoopoepg** [comment]
logger tests disables DM

**Comment by yosefe** [comment]
ok, pls make the needed fix in mlx5 in this PR as pre-condition to enable DM, otherwise would not be possible to debug UCX with DM

**Comment by hoopoepg** [comment]
picked into https://github.com/openucx/ucx/pull/2425

**Pull request closed without merging**